dm/docs/RFCS/20211130_enhanced_pre_checker.md

```diff
@@ -0,0 +1,86 @@
+    - auto_increment_ID
+2. We can adjust the concurrency by table numbers.
+
+### Optimize some check
+
+1. Auto_increment_ID only be checked in sharding mode.
+    - If table exists auto increment ID, report a warning to user and tell them the method that can resolve the PK/UK conflict;
+        1. If you set PK to AUTO INCREMENT, you must make sure that the primary key in sharding tables is not duplicated;
+        2. If sharding tables have duplicated PK, you can refer to [document](https://docs.pingcap.com/tidb-data-migration/stable/shard-merge-best-practices#handle-conflicts-of-auto-increment-primary-key).
+    - And if they have finished the resolving, such as manually created the table and removed AUTO INCREMENT, we should not report the warning.
+2. Dump_privilege will check different privileges according to different [consistency](https://docs.pingcap.com/tidb/stable/dumpling-overview#adjust-dumplings-data-consistency-options) on source.
+    - For all consistency, we will check
+        - SELECT (only INFORMATION_SCHEMA's tables and dump tables)
+    - For flush consistency:
+        - RELOAD (global)
+    - For lock consistency:
+        - LOCK TABLES (only dump tables)
+3. Add OnlineDDLChecker to check if a DDL of tables in allow list exists in online-ddl stage when DM task is all mode and online-ddl is true.
+4. Enhance schema_of_shard_tables. 
+    - If a task has passed the pre-checking when starting and exited, DM should keep the consistency during the task running. So we **don't check it** when restart the task.
+    - If there does not exist checkpoints:
+        - For all/full mode (pessimistic task): we keep **the original check**;
+        - For all/full mode (optimistic task): we check whether the shard tables schema meets **the definition of [Optimistic Schema Compatibility](20191209_optimistic_ddl.md)**. If that meets, we can create tables by the compatible schema in the dump stage.
+        - For incremental mode: **not check** the sharding tables’ schema, because the table schema obtained from show create table is not the schema at the point of binlog.
+5. Version checker will report a warning in the following cases:
+    - MySQL < 5.6.0
+    - MySQL >= 8.0.0
+    - Mariadb
+
+### Restrict user usage
+1. Remove all `ignore_check_items` settings from the [document](https://docs.pingcap.com/tidb-data-migration/stable/precheck#disable-checking-items). If the following items are detected to be set in the configuration, a warning will be reported.
+2. If task is full/all mode, the following items will be forced to check (correspondingly, it will not be check in increment mode):
+    - dump_privilege
+    - schema_of_shard_tables(only for sharding mode)
+3. If task is increment/all mode, the following items will be forced to check (correspondingly, it will not be check in full mode):
+    - replication_privilege
+    - server_id
+    - binlog_enable
+    - binlog_format
+    - binlog_row_image
+    - online_ddl(new added)
```

Comment:
need check `Binlog_Do_DB`

Reply:
add it in https://github.com/pingcap/tiflow/pull/3674/commits/2ea1a63c2d7cedfb2dda6e9059be15807f674acf